code-dot-org/code-dot-org
# Fix empty states for v1 teacher dashboard

With the migration to react-router-dom v6, I accidentally broke the empty states when a user was looking at a section without students or without students who had made progress (showing that they had no curriculum). This is because react-router-dom changed routes from being displayed in the order they were defined to a tree structure with the highest matching depth element being displayed instead.

e.g.
```
<Route path='/' element={<ElementNoPath />} >
<Route path='/test' element={<ElementTest />} >
```
Before v6, the above would show the ElementNoPath for url `/test` because no path was defined above test path. After v6, it selects `ElementTest` because the `/test` path is more specific than `/`.

This PR fixes that issue by removing the logic to select the empty state from the router entirely and instead has a function that is applied to elements that determines whether to show the empty state or not.

Functional changes:
* Adds empty state to `/standards_report`. This can only be reached by the progress page, so it should be fine

Other changes:
* Extra base route (`'/*'`) in show is unnecessary. Removed.
* Router => BrowserRouter. This is just clearer since we were importing with `as`. It confused me a little bit when I first started looking at this and is unnecessary.
* Add a better error message to `/standards_report` when not seen from an appropriate path and not enough information. Ideally, we would remove this as a path in the router and instead put it into the progress page. Since we did not port the `standards` tab to v2 progress page, we can most likely instead just remove the tab entirely when we remove the v1.
  * See https://codedotorg.slack.com/archives/C06ERUABG01/p1716312165751379
  * This was because this is the second time where I've gone to the url and had to go digging to see why there was a null pointer exception.

## Links
[TEACH-1247](https://codedotorg.atlassian.net/browse/TEACH-1247)




## Testing story





## Deployment strategy

## Follow-up work



## Privacy



## Security



## Caching

## PR Checklist:
